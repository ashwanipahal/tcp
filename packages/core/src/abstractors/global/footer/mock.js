export default {
  footer_top: {
    email_sms_signup: {
      url: '/',
      title: 'Get $10 off by signing up for email or text',
      text: 'GET <span class="l-color-primary">$10 OFF</span> BY SIGNING UP FOR EMAIL OR TEXT',
      target: 'email_signup_popup',
    },
    refer_a_fried: {
      url: '/',
      title: 'Refer a friend and earn another $10',
      text: 'REFER A FRIEND AND EARN ANOTHER <span class="l-color-primary">$10</span>',
      target: '',
    },
    social_media_links: [
      {
        url: 'http://facebook.com/childrensplace',
        title: 'Facebook',
        target: '_blank',
        icon_class: 't-icon-facebook',
      },
      {
        url: 'http://twitter.com/childrensplace',
        title: 'Twitter',
        target: '_blank',
        icon_class: 't-icon-twitter',
      },
      {
        url: 'http://instagram.com/childrensplace',
        title: 'Instagram',
        target: '_blank',
        icon_class: 't-icon-instagram',
      },
      {
        url: 'http://pinterest.com/childrensplace',
        title: 'Pinterest',
        target: '_blank',
        icon_class: 't-icon-pinterest',
      },
    ],
  },
  footer_middle: {
    mpr: {
      link: {
        url: '/',
        image_alt: 'My Place Rewards image',
        image_url: '/',
        title: 'My Place rewards title',
        text: 'My Place Rewards',
        target: '_blank',
      },
      sub_links: [
        {
          url: '/',
          title: 'Check Point Balance title',
          text: 'Check Point Balance',
          target: '',
        },
        {
          url: '/',
          title: 'Redeem rewards title',
          text: 'Redeem Rewards',
          target: '',
        },
        {
          url: '/',
          title: 'Member benefits title',
          text: 'Member Benefits',
          target: '',
        },
      ],
    },
    mpr_cc: {
      link: {
        url: '/',
        image_alt: 'My Place Rewards Credit Card image',
        image_url: '/',
        title: 'My Place rewards credit card title',
        text: 'My Place Rewards Credit Card',
        target: '_blank',
      },
      sub_links: [
        {
          url: '/',
          title: 'Learn more title',
          text: 'Learn More',
          target: '',
        },
        {
          url: '/',
          title: 'Apply now title',
          text: 'Apply Now',
          target: '',
        },
        {
          url: '/',
          title: 'Pay your bill title',
          text: 'Pay Your Bill',
          target: '',
        },
        {
          url: '/',
          title: 'Manage your account title',
          text: 'Manage Your Account',
          target: '',
        },
      ],
    },
    nav_links: [
      {
        items: [
          {
            text: 'HELP CENTER',
            links: [
              {
                url: '/',
                text: 'FAQs',
                title: 'Frequently Asked Questions',
              },
              {
                url: '/',
                text: 'Track Order',
                title: 'Track Order Title',
              },
              {
                url: '/',
                text: 'Return Policy',
                title: 'Return Policy Title',
              },
            ],
          },
        ],
      },
      {
        items: [
          {
            text: 'SHOPPING',
            links: [
              {
                url: '/',
                text: 'Log Out',
                title: 'Log Out Title',
              },
              {
                url: '/',
                text: 'Coupons',
                title: 'Coupons Title',
              },
              {
                url: '/',
                text: 'Store Locator',
                title: 'Store Locator Title',
              },
            ],
          },
        ],
      },
      {
        items: [
          {
            text: 'ABOUT US',
            links: [
              {
                url: '/',
                text: 'Public Relations',
                title: 'Public Relations Title',
              },
              {
                url: '/',
                text: 'Investor Relations',
                title: 'Investor Relations Title',
              },
              {
                url: '/',
                text: 'Careers',
                title: 'Careers Title',
              },
            ],
          },
        ],
      },
    ],
  },
  footer_bottom: {
    copyrights: {
      text: "<p>&copy; 2019 The Children's Place | Big fashion, little prices</p>",
    },
    legal_links: [
      {
        url: 'https://www.childrensplace.com/us/help-center/#termsAndConditionsli',
        title: 'Terms and Conditions title',
        text: 'Terms and Conditions',
        locator: 'global_T&Clink',
        target: '',
      },
      {
        url: 'https://www.childrensplace.com/us/help-center/#privacyPolicySectionli',
        title: 'Privacy Policy title',
        text: 'Privacy Policy',
        locator: 'global_privacypolicylink',
        target: '',
      },
      {
        url: 'https://www.childrensplace.com/us/help-center/#cookiePolicySectionli',
        title: 'Cookie Policy title',
        text: 'Cookie Policy',
        locator: 'global_cookiepolicylink',
        target: '',
      },
      {
        url: 'https://www.childrensplace.com/us/content/supply-chain',
        title: 'California supply chain act title',
        text: 'California Supply Chains Act',
        locator: 'global_californiaSCAlink',
        target: '',
      },
      {
        url: 'https://www.childrensplace.com/us/sitemap',
        title: 'Site Map title',
        text: 'Site Map',
        locator: 'global_siteMaplink',
        target: '',
      },
    ],
  },
};
