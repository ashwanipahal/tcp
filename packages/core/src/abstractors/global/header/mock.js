export default {
  header_top_nav: {
    brand_tabs: [
      {
        url: '/',
        alt: "The Children's Place",
        title: "The Children's Place",
        target: '',
        class: 'tcp-logo-medium',
      },
      {
        url: 'https://www.gymboree.com/',
        alt: 'Gymboree',
        title: 'Gymboree',
        target: '_blank',
        logo_class: 'gymboree-logo-medium',
      },
    ],
    promo_message_wrapper: [
      {
        url: '/static/promo',
        text: '<b>FREE SHIPPING EVERY DAY!</b> No minimum purchase required 1.',
        title: 'FREE SHIPPING EVERY DAY! No minimum purchase required 1.',
        target: '_blank',
        class: 'promo_message',
      },
      {
        url: '/static/promo2',
        text: '<b>FREE SHIPPING EVERY DAY!</b> No minimum purchase required 2.',
        title: 'FREE SHIPPING EVERY DAY! No minimum purchase required 2.',
        target: '_blank',
        class: 'promo_message',
      },
      {
        url: '/static/promo3',
        text: '<b>FREE SHIPPING EVERY DAY!</b> No minimum purchase required 3.',
        title: 'FREE SHIPPING EVERY DAY! No minimum purchase required 3.',
        target: '_blank',
        class: 'promo_message',
      },
    ],
  },
  header_promo_area: {
    promo_text_banners: [
      {
        url: '/banner/url1',
        promo_text: {
          primary: {
            text: '<b>NEED IT NOW?</b>',
            color: '#1a1a1a',
          },
          secondary: {
            text: 'Buy online, pickup in store.',
            color: '#1a1a1a',
          },
        },
        title: 'NEED IT NOW? Buy online, pickup in store.',
        target: '',
        class: 'promo_icon_clock',
      },
      {
        url: '/banner/url2',
        promo_text: {
          primary: {
            text: '<b>EARN PLACE CASH!</b>',
            color: '#7dc24c',
          },
          secondary: {
            text: 'Get 10$ for every 20$ spent today.',
            color: '#1a1a1a',
          },
        },
        title: 'EARN PLACE CASH! Get 10$ for every 20$ spent today.',
        target: '',
        class: 'promo_icon_dollar',
      },
      {
        url: '/node/modal/rr1',
        promo_text: {
          primary: {
            text: '<b>SIGN UP AND GET 10$ OFF!</b>',
            color: '#1a1a1a',
          },
          secondary: {
            text: '<u>ENTER YOUR EMAIL.</u>',
            color: '#1a1a1a',
          },
        },
        title: 'SIGN UP AND GET 10$ OFF! ENTER YOUR EMAIL.',
        target: '',
        class: 'promo_icon_at',
      },
    ],
    promo_loyalty_banners: [
      {
        url: '/banner/url',
        image_url: 'http://image1.sm/url',
        image_alt: 'Banner Image 1 Alt text attribute val',
        title: 'Banner Image 1 Title attribute val',
        text: '<b>Banner 1 Text val</b>',
        target: '',
        class: 'promo_loyalty_bonus',
      },
      {
        url: '/banner/url',
        image_url: 'http://image2.sm/url',
        image_alt: 'Banner Image 2 Alt text attribute val',
        title: 'Banner Image 2 Title attribute val',
        text: '<b>Banner 2 Text val</b>',
        target: '',
        class: 'promo_loyalty_reward',
      },
    ],
  },
};
