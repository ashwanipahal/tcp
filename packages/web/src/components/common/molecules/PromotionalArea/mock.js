export default {
  promo_message_wrapper: [
    {
      id: '1',
      richText: {
        text: '<b>FREE SHIPPING EVERY DAY!</b> No minimum purchase required 1.',
        title: 'FREE SHIPPING EVERY DAY! No minimum purchase required 1.',
      },
      link: { url: '/static/promo', target: '_blank' },
      class: 'promo_message',
    },
    {
      id: '2',
      richText: {
        text: '<b>FREE SHIPPING EVERY DAY!</b> No minimum purchase required 2.',
        title: 'FREE SHIPPING EVERY DAY! No minimum purchase required 2.',
      },
      link: { url: '/static/promo2', target: '_blank' },
      class: 'promo_message',
    },
    {
      id: '3',
      richText: {
        text: '<b>FREE SHIPPING EVERY DAY!</b> No minimum purchase required 3.',
        title: 'FREE SHIPPING EVERY DAY! No minimum purchase required 3.',
      },
      link: {
        url: '/static/promo3',
        target: '_blank',
      },
      class: 'promo_message',
    },
  ],
};
