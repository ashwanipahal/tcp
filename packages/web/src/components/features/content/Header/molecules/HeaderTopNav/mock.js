export default {
  dataTopNav: {
    composites: {
      brand_tabs: [
        {
          url: '/',
          title: 'TCP Link',
          target: '',
          external: 0,
          class: 'header__brand-tab--tcp',
          __typename: 'LinkClass',
        },
        {
          url: 'https://www.gymboree.com/',
          title: 'Gymboree Link',
          target: '_blank',
          external: 0,
          class: 'header__brand-tab-gymboree',
          __typename: 'LinkClass',
        },
      ],
      promo_message_wrapper: [
        {
          richText: {
            text: '<b>FREE SHIPPING EVERY TEST DAY!</b> No minimum purchase required.',
            __typename: 'Text',
          },
          link: {
            url: '/static/promo1',
            title: '',
            target: '',
            external: 0,
            __typename: 'Link',
          },
          __typename: 'PromoMessage',
        },
        {
          richText: {
            text: '<b>FREE SHIPPING EVERY DAY!</b> This is banner2',
            __typename: 'Text',
          },
          link: {
            url: '/static/promo2',
            title: '',
            target: '',
            external: 0,
            __typename: 'Link',
          },
          __typename: 'PromoMessage',
        },
        {
          richText: {
            text: '<b>FREE SHIPPING EVERY DAY!</b> Banner 3 comes now',
            __typename: 'Text',
          },
          link: {
            url: '/static/promo3',
            title: '',
            target: '',
            external: 0,
            __typename: 'Link',
          },
          __typename: 'PromoMessage',
        },
      ],
      __typename: 'Composite',
    },
  },
};
