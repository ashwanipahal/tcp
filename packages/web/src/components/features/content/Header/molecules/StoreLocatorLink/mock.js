export default {
  currentStore: {
    basicInfo: {
      id: '111421',
      storeName: 'steinway street',
      isDefault: false,
      address: {
        addressLine1: '31-53 steinway street',
        city: 'astoria',
        state: 'NY',
        country: 'US ',
        zipCode: '11103',
      },
      phone: '(718) 956-1999',
      coordinates: {
        lat: 40.76004,
        long: -73.91805,
      },
      x_stloc: {
        metaKeyword: '',
        metaDescription: '',
        pageTitle: '',
      },
    },
    hours: {
      regularHours: [
        {
          dayName: 'FRIDAY',
          openIntervals: [
            {
              fromHour: '2019-01-04 10:00:00',
              toHour: '2019-01-04 20:00:00',
            },
          ],
          isClosed: false,
        },
        {
          dayName: 'SATURDAY',
          openIntervals: [
            {
              fromHour: '2019-01-05 10:00:00',
              toHour: '2019-01-05 20:00:00',
            },
          ],
          isClosed: false,
        },
        {
          dayName: 'THREE KINGS DAY',
          openIntervals: [
            {
              fromHour: '2019-01-06 11:00:00',
              toHour: '2019-01-06 18:00:00',
            },
          ],
          isClosed: false,
        },
        {
          dayName: 'MONDAY',
          openIntervals: [
            {
              fromHour: '2019-01-07 10:00:00',
              toHour: '2019-01-07 20:00:00',
            },
          ],
          isClosed: false,
        },
        {
          dayName: 'TUESDAY',
          openIntervals: [
            {
              fromHour: '2019-01-08 10:00:00',
              toHour: '2019-01-08 20:00:00',
            },
          ],
          isClosed: false,
        },
        {
          dayName: 'WEDNESDAY',
          openIntervals: [
            {
              fromHour: '2019-01-09 10:00:00',
              toHour: '2019-01-09 20:00:00',
            },
          ],
          isClosed: false,
        },
        {
          dayName: 'THURSDAY',
          openIntervals: [
            {
              fromHour: '2019-01-10 10:00:00',
              toHour: '2019-01-10 20:00:00',
            },
          ],
          isClosed: false,
        },
        {
          dayName: 'FRIDAY',
          openIntervals: [
            {
              fromHour: '2019-01-11 10:00:00',
              toHour: '2019-01-11 20:00:00',
            },
          ],
          isClosed: false,
        },
        {
          dayName: 'SATURDAY',
          openIntervals: [
            {
              fromHour: '2019-01-12 10:00:00',
              toHour: '2019-01-12 20:00:00',
            },
          ],
          isClosed: false,
        },
      ],
      holidayHours: [],
      regularAndHolidayHours: [],
    },
    features: {
      storeType: 'Retail Store',
      mallType: 'STREET LOCATION',
      entranceType: 'EXTERNAL',
      isBopisAvailable: false,
    },
  },
};
