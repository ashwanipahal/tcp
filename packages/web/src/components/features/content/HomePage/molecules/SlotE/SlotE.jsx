import dynamic from 'next/dynamic';
import React from 'react';
import { PropTypes } from 'prop-types';

const returnModule = mod => mod.default;

const SlotE = dynamic({
  modules: () => ({
    ModuleD: () => import('@tcp/core/src/components/common/molecules/ModuleD').then(returnModule),
    ModuleH: () => import('@tcp/core/src/components/common/molecules/ModuleH').then(returnModule),
    ModuleK: () => import('@tcp/core/src/components/common/molecules/ModuleK').then(returnModule),
    ModuleL: () => import('@tcp/core/src/components/common/molecules/ModuleL').then(returnModule),
    ModuleA: () => import('@tcp/core/src/components/common/molecules/ModuleA').then(returnModule),
  }),
  render: (slotProps, { ModuleD, ModuleH, ModuleK, ModuleL, ModuleA }) => {
    const { name } = slotProps;
    switch (name) {
      case 'moduleA':
        return <ModuleA {...slotProps} />;
      case 'moduleD':
        return <ModuleD {...slotProps} />;
      case 'moduleH':
        return <ModuleH {...slotProps} />;
      case 'moduleK':
        return <ModuleK {...slotProps} />;
      case 'moduleL':
        return <ModuleL {...slotProps} />;
      default:
        return null;
    }
  },
});

SlotE.defaultProps = {
  name: null,
};

SlotE.propTypes = {
  name: PropTypes.string,
};

export default SlotE;
