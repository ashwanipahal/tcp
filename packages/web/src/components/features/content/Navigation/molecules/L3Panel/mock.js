export default {
  shopAllLabel: 'Shop All',
};
