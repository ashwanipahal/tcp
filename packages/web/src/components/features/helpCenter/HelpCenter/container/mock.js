export default {
  Layouts: {
    helpCenterPage: {
      slots: [
        {
          name: 'slot_1',
          moduleName: 'helpCenterHome',
          contentId: 'test',
        },
      ],
      __typename: 'LayoutInfo',
    },
  },
  HelpCenterReducer: {
    test: [
      {
        subSections: null,
        leafLink: {
          url: '/content/',
          text: 'Help Center Home',
          title: 'Help Center Home',
          target: '',
          external: 0,
        },
      },
      {
        subSections: null,
        leafLink: {
          url: '/content/',
          text: 'My Place Rewards Credit Card',
          title: 'My Place Rewards Credit Card',
          target: '',
          external: 0,
        },
      },
      {
        subSections: null,
        leafLink: {
          url: '/content/',
          text: 'Coupons and Promotions',
          title: 'Coupons and Promotions',
          target: '',
          external: 0,
        },
      },
      {
        subSections: null,
        leafLink: {
          url: '/content/',
          text: 'Help With My Order',
          title: 'Help With My Order',
          target: '',
          external: 0,
        },
      },
      {
        subSections: null,
        leafLink: {
          url: '/content/',
          text: 'Online Ordering',
          title: 'Online Ordering',
          target: '',
          external: 0,
        },
      },
      {
        subSections: null,
        leafLink: {
          url: '/content/',
          text: 'Policies',
          title: 'Policies',
          target: '',
          external: 0,
        },
      },
      {
        subSections: null,
        leafLink: {
          url: '/content/',
          text: 'Terms and Conditions',
          title: 'Terms and Conditions',
          target: '',
          external: 0,
        },
      },
      {
        subSections: null,
        leafLink: {
          url: '/content/',
          text: 'Mobile Terms and Conditions ',
          title: 'Mobile Terms and Conditions ',
          target: '',
          external: 0,
        },
      },
      {
        subSections: null,
        leafLink: {
          url: '/content/',
          text: 'Shipping to the U.S.',
          title: 'Shipping to the U.S.',
          target: '',
          external: 0,
        },
      },
      {
        subSections: null,
        leafLink: {
          url: '/content/',
          text: 'Ordering From Canada',
          title: 'Ordering From Canada',
          target: '',
          external: 0,
        },
      },
      {
        subSections: null,
        leafLink: {
          url: '/content/',
          text: 'Tools',
          title: 'Tools',
          target: '',
          external: 0,
        },
      },
      {
        subSections: null,
        leafLink: {
          url: '/content/',
          text: 'Contact Us',
          title: 'Contact Us',
          target: '',
          external: 0,
        },
      },
    ],
  },
};
