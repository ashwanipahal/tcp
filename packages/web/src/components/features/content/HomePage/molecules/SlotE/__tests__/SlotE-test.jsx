import React from 'react';
import { shallow } from 'enzyme';
import mock from '@tcp/core/src/services/abstractors/common/moduleL/mock';
import SlotE from '../SlotE';

describe('SlotB component', () => {
  it('SlotA component renders correctly', () => {
    const component = shallow(<SlotE {...mock} />);
    expect(component).toMatchSnapshot();
  });
});
