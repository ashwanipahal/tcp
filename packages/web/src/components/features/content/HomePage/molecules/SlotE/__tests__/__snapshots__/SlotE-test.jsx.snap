// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlotE component SlotE component renders correctly 1`] = `
<loading
  error={null}
  isLoading={true}
  pastDelay={false}
  retry={[Function]}
  timedOut={false}
/>
`;
